src/modules/AdvancedPaste/AdvancedPaste/Helpers/LogEvent.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json;
+using AdvancedPaste.Models.KernelQueryCache;
+
+namespace AdvancedPaste.Helpers
+{
+    public class LogEvent
+    {
+        public LogEvent(object message)
+        {
+            this.message = message;
+        }
+
+        private object message;
+
+        public string ToJsonString() => JsonSerializer.Serialize(this, AdvancedPasteJsonSerializerContext.Default.PersistedCache);
```

Comment:
```suggestion
        public string ToJsonString() => JsonSerializer.Serialize(this, AdvancedPasteJsonSerializerContext.Default.LogEvent);
```

Also, I believe the `LogEvent` class should be more defined. 

Since we know `telemetryEvent.CacheUsed`, `telemetryEvent.IsSavedQuery`, `telemetryEvent.PromptTokens`, `telemetryEvent.CompletionTokens`, `telemetryEvent.ModelName`, `telemetryEvent.ActionChain` are the properties passed into the anonymous object, defining them in the class will make it cleaner and less prone to potential issues.

Reply:
Sure, I agree.

Reply:
Let me consider how to make this interface easy to use.

Reply:
@snickler 
I've updated the LogEvent define. Could you please take a look?